Allow queries with "tag=*" tag filters in expressions

wicknicks opened the pull request:
This was previously not supported because if you need this, you can use the regular TSDB interface. But with some graphite functions (for example, highestMax), this feature would be useful.

wicknicks committed:
Allow queries with tag=* tag filters in expressions

This was previously not supported because if you need this,
you can use the regular TSDB interface. Pushing this to a
different branch only to keep a note of the change required.

wicknicks commented:
The only change is in parser.jj. The rest of the files are generated by javacc.

absaxena commented:
UnZb

Sent from my iPhone

On Jun 30, 2015, at 1:32 PM, "Arjun Satish" <notifications@github.com<mailto:notifications@github.com>> wrote:

This was previously not supported because if you need this, you can use the regular TSDB interface. But with some graphite functions (for example, highestMax), this feature would be useful.

---

You can view, comment on, or merge this pull request online at:

  https://github.com/turn/opentsdb/pull/1

Commit Summary
-   Allow queries with tag=\* tag filters in expressions

File Changes
-   M src/tsd/expression/parser/SyntaxChecker.javahttps://github.com/turn/opentsdb/pull/1/files#diff-0 (41)
-   A src/tsd/expression/parser/SyntaxCheckerTest.javahttps://github.com/turn/opentsdb/pull/1/files#diff-1 (31)
-   M src/tsd/expression/parser/SyntaxCheckerTokenManager.javahttps://github.com/turn/opentsdb/pull/1/files#diff-2 (28)
-   M src/tsd/expression/parser/parser.jjhttps://github.com/turn/opentsdb/pull/1/files#diff-3 (2)
-   A src/tsd/expression/parser/run_javacc.shhttps://github.com/turn/opentsdb/pull/1/files#diff-4 (2)

Patch Links:
-   https://github.com/turn/opentsdb/pull/1.patch
-   https://github.com/turn/opentsdb/pull/1.diff

## 

Reply to this email directly or view it on GitHubhttps://github.com/turn/opentsdb/pull/1.

wicknicks committed:
Adding more features to the alias function.

We can now alias by tag values. A string template can be sent as a parameter
which will be used in conjunction with the metric tags to create the alias
string. A tag is indicated as '@tagk', which will be replaced with the 'tagv'
value while formatting.

Signed-off-by: Arjun Satish <arjun.satish@turn.com>
